### PR TITLE
add csv2 lib

### PR DESCRIPTION
**Pull request by wanghenshui**

### CODE REVIEW

**Comment by wanghenshui** [comment]
ping @waruqi
please let workflow run

File: packages/c/csv2/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package("csv2")
+    set_urls("https://github.com/p-ranav/csv2.git")
+    set_homepage("https://github.com/p-ranav/csv2")
+    set_description("A CSV parser library")
+    set_license("MIT")
```
Comment (waruqi) [inline packages/c/csv2/xmake.lua:5]: please move set_urls to here and add a newline first

Comment (waruqi) [reply packages/c/csv2/xmake.lua:5]: please add `https://github.com/p-ranav/csv2/archive/refs/tags/v0.1.tar.gz`

```lua
add_urls("https://github.com/p-ranav/csv2/archive/refs/tags/$(version).tar.gz",
       "https://github.com/p-ranav/csv2.git")
```

Comment (waruqi) [reply packages/c/csv2/xmake.lua:5]: please add `https://github.com/p-ranav/csv2/archive/refs/tags/v0.1.tar.gz`

```lua
add_urls("https://github.com/p-ranav/csv2/archive/refs/tags/$(version).tar.gz",
       "https://github.com/p-ranav/csv2.git")
```

File: packages/c/csv2/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    add_versions("v0.1", "a20992f7a1b75215609037c4cc66f690e8691aab")
```
Comment (waruqi) [inline packages/c/csv2/xmake.lua:6]: `a20992f7a1b75215609037c4cc66f690e8691aab ` it should be the shasum256 of tar.gz

File: packages/c/csv2/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    on_install(function (package)
+        os.cp("include/csv2", package:installdir("include"))
+    end)
```
Comment (waruqi) [inline packages/c/csv2/xmake.lua:9]: please add newline

File: packages/c/csv2/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package("csv2")
```
Comment (waruqi) [inline packages/c/csv2/xmake.lua:1]: add `set_kind("library", {headeronly = true})` mark as headeronly library

**Comment by wanghenshui** [comment]
ping @waruqi
please let workflow run

**Comment by waruqi** [comment]
Thanks!

**Pull request merged**